utils/misc.ml

```diff
@@ -181,15 +181,50 @@ let may_map = Stdlib.Option.map
 
 (* File functions *)
 
+let find_cache = Hashtbl.create 16
+let find_cache_cwd = ref (Sys.getcwd ())
+let reset_find_cache () =
+  Hashtbl.clear find_cache
+
+let really_exists dir name =
+  let dir = if dir = "" then Filename.current_dir_name else dir in
+  let () =
+    if Filename.is_relative dir then
+      let cwd = Sys.getcwd () in
+      if !find_cache_cwd <> cwd then begin
+        Hashtbl.clear find_cache;
+        find_cache_cwd := cwd
```

Comment:
Why not use `let dir = Filename.concat cwd dir in` in that case rather than clear the cache?